### DISCUSSION TITLE
I can't figure out how to modify the landscape project to get it compiled using airsim..

### DISCUSSION THREAD

**Discussion by youthfrog** [microsoft/AirSim#3388] [Support Q&A] [answered]
unreal engine 4.24.3 + vs 2019
Previously, I was trapped in the problems of eigen3. 
Then, I delete the empty folder(eigen3), and ran the build.cmd under the folder of airsim.
After that, I followed the guidance in the Internet,
change the original code of landscape project from:
`{
	"FileVersion": 3,
	"EngineAssociation": "4.24",
	"Category": "Samples",
	"Description": "",
	"Modules": [
		{
			"Name": "finalproject",
			"Type": "Runtime",
			"LoadingPhase": "Default"
		}
	],
	"TargetPlatforms": [
		"MacNoEditor",
		"WindowsNoEditor"
	],
	"EpicSampleNameHash": "1226740271"
}` 
to:
`{
	"FileVersion": 3,
	"EngineAssociation": "4.24",
	"Category": "Samples",
	"Description": "",
	"Modules": [
		{
			"Name": "finalproject",
			"Type": "Runtime",
			"LoadingPhase": "Default",
			"AdditionalDependencies": [
				"AirSim"
			]
		}
	],
	"TargetPlatforms": [
		"MacNoEditor",
		"WindowsNoEditor"
	],
	"Plugins": [
		{
			"Name": "AirSim",
			"Enabled": true
		}
	],
	"EpicSampleNameHash": "1226740271"
}`
I copied the plugins from airsim folder to the project folder(I named its folder: finalproject).
After that, I generated project files and double clicked the unreal engine project.
Now error occurs:
`the following modules are missing or built with a different engine version:
Airsim
Would you like to rebuild them now?`
I click yes and find `finalproject could not be compiled. Try rebuilding from source manually.` 
I check the logs of this failure, and found:
`  [1/5] Module.AirSim.cpp
  D:/plane/finalproject/Plugins/AirSim/Source/SimMode/SimModeBase.cpp(222): error C2065: ??FObjectProperty??: δ?????ı?ʶ??
  D:/plane/finalproject/Plugins/AirSim/Source/SimMode/SimModeBase.cpp(222): error C2065: ??sun_prop??: δ?????ı?ʶ??
  D:/plane/finalproject/Plugins/AirSim/Source/SimMode/SimModeBase.cpp(222): error C2065: ??CastFieldChecked??: δ?????ı?ʶ??
  D:/plane/finalproject/Plugins/AirSim/Source/SimMode/SimModeBase.cpp(222): error C2065: ??FObjectProperty??: δ?????ı?ʶ??
  D:/plane/finalproject/Plugins/AirSim/Source/SimMode/SimModeBase.cpp(223): error C2065: ??sun_prop??: δ?????ı?ʶ??
  D:/plane/finalproject/Plugins/AirSim/Source/WorldSimApi.cpp(183): warning C4457: ??position?????????????˺???????
LogInit: Warning: Still incompatible or missing module: AirSim`
I cannot figure this out! Please help me!

**Comment by rajat2004** [comment, marked as answer]
Recommended UE version is 4.25 since a few months. PR https://github.com/microsoft/AirSim/pull/3268 changed the line which is giving the error, you could either upgrade to 4.25 or if not possible, change the specific line to the older version, and it should compile. There have been some asset changes as well which might not work

**Reply by youthfrog** [reply]
Thank you so much for your reply, my problem is solved, thank you!
I can finally start the next steps!
Oh! I'm so excited!

**Reply by AlexanderZ3** [reply]
Hi man, same questions, could you share with me how to solve the question? I have copied the Plugins folder from the AirSim to Landscape, and edit the landscape.uproject as tutorials shown(same as you mentioned), but the landscape project cannot be compiled.  I tried many times and it still does not work..... hope you can give me some suggestions

**Reply by ahmed-elsaharti** [reply]
Hi @AlexanderZ3 , Are you getting the same compilation error above? if that's the case then as Rajat mentioned, the cleanest solution is to upgrade to Unreal Engine 4.25. If you're unable to upgrade, try #3404

**Reply by AlexanderZ3** [reply]
> Hi @AlexanderZ3 , Are you getting the same compilation error above? if that's the case then as Rajat mentioned, the cleanest solution is to upgrade to Unreal Engine 4.25. If you're unable to upgrade, try #3404

Thanks, but the landscape environment only supports UE4.24, is it right?

**Reply by ahmed-elsaharti** [reply]
@AlexanderZ3 it should work with 4.25 too AFAIK

**Reply by AlexanderZ3** [reply]
> @AlexanderZ3 it should work with 4.25 too AFAIK

![image](https://user-images.githubusercontent.com/29619650/109096292-3ed66f00-7758-11eb-8a3e-72bcdb559268.png)

**Reply by ahmed-elsaharti** [reply]
@AlexanderZ3  
![image](https://user-images.githubusercontent.com/49626509/109216464-2d9e5a00-777a-11eb-9b81-9c3bb93bcfd0.png)

**Reply by youthfrog** [reply]
@AlexanderZ3 
Sorry, I was late in replying. 
First of all, you can only use the version before 4.24 at the beginning. If you want to use 4.25, you can try to install the landscape first, uninstall the corresponding unreal engine version, and install the 4.25 version of unreal engine, then click the corresponding .uproject file to get an option for you to choose 'the version' of the unreal engine.
The above is actually a solution using version 4.25. If you still want to be in the lower version, just modify the corresponding statement in the code of  PR #3268.

**Reply by AlexanderZ3** [reply]
@youthfrog 
Thanks a lot! I will try this solution in UE4.25 environment. I just tried the UE4.24 version environment, it works well for the Landscape Mountain project.

**Reply by AlexanderZ3** [reply]
> @AlexanderZ3
> ![image](https://user-images.githubusercontent.com/49626509/109216464-2d9e5a00-777a-11eb-9b81-9c3bb93bcfd0.png)

@ahmed-elsaharti  Thanks! You are right.
Now I run the project successfully in the UE 4.24 version, my previous problem is that I built the Aisim in the UE4.25 version. So that even though I run the landscape in 4.24, but it still did not work. After that, when the Airsim is built in the correct UE version, it works.

Thanks again, guys! Your help makes alone research journey not lonely.

**Reply by rajat2004** [reply]
The latest master should work in both 4.24 and 4.25 as well since https://github.com/microsoft/AirSim/pull/3404 has been merged
